datafusion/physical-expr/src/utils/guarantee.rs

```diff
@@ -93,18 +93,18 @@ impl LiteralGuarantee {
     /// Create a new instance of the guarantee if the provided operator is
     /// supported. Returns None otherwise. See [`LiteralGuarantee::analyze`] to
     /// create these structures from an predicate (boolean expression).
-    fn try_new<'a>(
+    fn new<'a>(
```

Comment:
Thanks @pgwhalen  -- this is a nice change and cleanup and makes sense to me

In order to make this change easier on downstream crates, can you please add back a version of `try_new()` that is marked "deprecated" with instructions on how to change the code  that just calls `new()`

Here is an example
https://github.com/apache/datafusion/blob/810e90802f695a60b733139da364ec344e46e43d/datafusion/execution/src/runtime_env.rs#L69-L72

Having a deprecation marking means that users will get a message from the compiler telling them what to do rather than having to track it down themselves

Reply:
Gotcha, I will add tonight!  I'd assumed it wasn't an issue because the function is not `pub` (it seems the main public interface to create LiteralGuarantees is `analyze`) - but I'm a little new to rust so I have some gaps in knowledge of how visibility works.

Reply:
> but I'm a little new to rust so I have some gaps in knowledge of how visibility works.

When in doubt I go to the docs.rs page:
https://docs.rs/datafusion/latest/datafusion/physical_expr/utils/struct.LiteralGuarantee.html

An in this case, I think you are right that this is a non pub method and therefore there is no downstream API implications.